docs/DeveloperGuide.md

```diff
@@ -299,6 +311,18 @@ list of lessons to view.
     * Cons: Operations are done with respect to the full list of lessons in the application, which means user cannot
       isolate the list of lessons of a student of choice to operate on.
 
+**Aspect: Recurrence rule for recurring lessons**
+
+* **Alternative 1 (current choice):** Only allow weekly recurrence
+    * Pros: Easy to implement and allow room for other features (such as detecting of clashes). Weekly recurring lessons are
+      common for private 1-to-1 tuition in Singapore.
+    * Cons: Users cannot set their own recurrence rule. Workaround for schedule-keeping possible through adding singular lessons
+      when needed.
+
+* **Alternative 2:** Allow user to define their own recurrence rule.
+    * Pros: Greater flexibility in planning lessons outside the regular weekly recurrence.
+    * Cons: Difficult to detect clashes with the endless possibilities.
+
```

Comment:
Please include why the current choice was chosen, e.g., lesson objects in a set in a person fits the 1-to-1 private home tutor more